lib/client/api_matrix.es6.js

```diff
@@ -53,6 +53,110 @@ foam.CLASS({
+          this.Browser.BROWSER_NAME,
+          this.mlang.GROUP_BY(
+            this.Browser.BROWSER_VERSION,
+            this.ArraySink.create())
+        )).then((group) => {
+          let browsers = {};
+          let browserNames = group.groupKeys;
+          for (let i = 0; i < browserNames.length; i++) {
+             browsers[browserNames[i]] = {};
+             let browserVersions = group.groups[browserNames[i]].groupKeys;
+             for (let j = 0; j < browserVersions.length; j++) {
+              browsers[browserNames[i]][browserVersions[j]] =
+                group.groups[browserNames[i]].groups[[browserVersions[j]]].a;
+             }
+          }
+          return browsers;
+        });
+      },
+    },
+    {
+      name: 'filterMatrix_',
+      documentation: `A post process of toMatrix, the result api toMatrix
+        will be filtered based on options.browserOptions and options.
+        numAvailable.`,
+      args: [
+        {
+          name: 'matrix',
+          typeName: 'JSON',
+          documentation: `JSON of the form:
+            {interfaceName: {apiName: {BrowserKey: true, ...} ...} ... } ...}`,
+        },
+        {
+          name: 'options',
+          typeName: 'JSON',
+          documentation: `An optional argument are used to filter the
+            matrix result:
+              searchKey: the result interface/API must contains this String
+                value, ignore cases.
+              browserOptions: An JSON of form {browserKey: boolean, ...}.
+                The returned result are filtered. So that only APIs that
+                whether it is available in this browser equals to
+                options[browser] are returned.
+                This option can be used to do queries such as finding
+                proprietary apis for a browser in a set of browsers.
+              numAvailable: an interger or interger Array that the web
+                interface whose number of available browser not included
+                in this array or not equal to this interger are filtered.
+                This option can be used to do queries such as finding apis
+                that exitsin at lease 3 browser vendors.`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `JSON of the form:
+            {interfaceName: {apiName: {BrowserKey: true, ...} ...} ... } ...}
+            This structure is easy for displaying a nested table.`,
+      },
+      code: function(matrix, options) {
+        options = options || {};
+        let browserOptions = options.browserOptions;
+        let numAvailable = options.numAvailable;
+        if (!browserOptions && !numAvailable) return matrix;
+        for (let webInterface in matrix) {
+          if (!matrix.hasOwnProperty(webInterface)) continue;
+          for (let api in matrix[webInterface]) {
+            if (!Object.prototype.hasOwnProperty
+              .call(matrix[webInterface], api)) continue;
```

Comment:
@shepheb FYI: This project and webidl-diff have been using one-line ifs when they're actually one line. Your comment here is constructive, but true-one-liners are okay.